src/utils/job_params_util.py

```diff
@@ -61,3 +79,276 @@ def get_resource_params_from_job_params(params):
         "gpu": gpu.to_dict(),
         "gpu_memory": gpu_memory.to_dict(),
     }
+
+
+class JobParams(object):
+    subclasses = {}
+
+    @classmethod
+    def register_subclass(cls, job_type):
+        def decorator(subclass):
+            cls.subclasses[job_type] = subclass
+            return subclass
+        return decorator
+
+    @classmethod
+    def create(cls, job_type, params, quota, metadata):
+        if job_type not in cls.subclasses:
+            raise ValueError("Bad job type %s" % job_type)
+        return cls.subclasses[job_type](params, quota, metadata)
+
+    def __init__(self, params, quota, metadata):
+        """Constructor for JobParams.
+
+        Args:
+            params: A dictionary of job parameters
+            quota: Resource quota
+            metadata: Resource metadata
+        """
+        self.job_id = params.get("jobId")
+        self.params = params
+        self.quota = quota
+        self.metadata = metadata
+
+        self.sku = None
+        self.gpu_limit = None
+        self.cpu_request = None
+        self.cpu_limit = None
+        self.memory_request = None
+        self.memory_limit = None
+
+        self.generate()
+
+    def generate(self):
+        self.gen_sku()
+        self.gen_gpu()
+        self.gen_cpu()
+        self.gen_memory()
+
+    def is_valid(self):
+        checklist = [
+            "sku",
+            "gpu_limit",
+            "cpu_request",
+            "cpu_limit",
+            "memory_request",
+            "memory_limit",
+        ]
+        for key in checklist:
+            if self.__dict__[key] is None:
+                return False
+        return True
+
+    def gen_sku(self):
+        sku = self.params.get("sku", "")
+        if sku != "":
+            self.sku = sku
+            return
+
+        sku_list = self.get_sku_list()
+        if len(sku_list) > 0:
+            self.sku = sku_list[0]
```

Comment:
Always choose the first one?

Reply:
We can define a configurable policy for this when we do global scheduling. For now, it's sufficient.